### DISCUSSION TITLE
Check if window is managed before resize operation, if it is, unfloat it first.

### DISCUSSION THREAD

**Discussion by szfkamil** [asmvik/yabai#2338] [General]
I have a proof of concept for a script that would

1. Check if property `manage` of focused window is `on` or `off`.
2. If it is off, perform the resize operation.
3. If it is `on` — thus it is managed, then float is first, then execute the re-size operation (only way to resize a window is when it is floated)

I think this can be used in `skhd` if the resize command is the argument. The argument would, as seen from my example below, be the command from after the script was invoked.

So:

`lalt - space     : yabai -m window --resize abs:700:1300` becomes something like: `lalt - space     :  yabai -m window check_if_floated.sh --resize abs:700:1300`

I'd like to know if,

a) by any chance, someone has already implemented this
b) if this is feasible at all, since I'm not a real programmer

**Comment by Bellavene** [comment]
A. Many times
B. It's ok

Here is an example
```
float_state=$(yabai -m query --windows --window | jq -r '.["is-floating"]')

if [ "$float_state" = "false" ]; then
   your command
else
   your command
fi
```

It can be transformed to one line like so:
```
float_state=$(yabai -m query --windows --window | jq -r '.["is-floating"]'); if [ "$float_state" = "false" ]; then; your command; else; your command; fi
```
